### PR TITLE
[Bugfix] Fix for #974

### PR DESCRIPTION
**Pull request by pgierz**
Hi, I have a quick fix for #974; we simply check if a file `/etc/os-release` exists, and grep for the appropriate release id if it does. If it does not the case statement reverts back to the default linux icon. This isn't so nice if you *do* in face have one of the pre-made icons (e.g. centos), but if the files are named inconsistently between different hosts, I don't see a good way of getting around that.

### CODE REVIEW

**Comment by bhilburn** [comment]
@pgierz - Great idea! The only thing I want to change about this PR is the use of `grep` - we are trying to remove as many calls to external programs as possible in the `next` codebase.

@onaforeignshore @rjorgenson - Any chance you can suggest a replacement pure-ZSH alternative to the use of `grep`, here? `=)`

**Comment by bhilburn** [comment]
Okay, I just realized that #988 fixed this in the `next` branch, hah. So, the issue is fixed in `next`, but not in `master`. Also, the fix indeed still uses `grep`.

If we are able to replace the use of `grep` in this PR, I'd like to merge this into master and then replace the same code in `next` with this change, as well.

File: functions/utilities.zsh [hunk 1]
```diff
@@ -102,7 +102,9 @@ case $(uname) in
       ;;
     Linux)
       OS='Linux'
-      os_release_id="$(grep -E '^ID=([a-zA-Z]*)' /etc/os-release | cut -d '=' -f 2)"
+      if [ -f /etc/os-release ]; then
+        os_release_id="$(grep -E '^ID=([a-zA-Z]*)' /etc/os-release | cut -d '=' -f 2)"
```
Comment (dritter) [inline functions/utilities.zsh]:
```suggestion
    [[ ${(f)"$((</etc/os-release) 2>/dev/null)"} =~ "ID=([A-Za-z]+)" ]] && os_release_id="${match[1]}"
```
This already includes the file check, so the currently enclosing if could be omitted too.

**Comment by dritter** [comment]
Hmm. Once again the travis tests on OSX hangs. Merging anyway. Thx @pgierz .

**Pull request merged**